lib/orderbook/MatchingEngine.ts

```diff
@@ -161,8 +161,18 @@ class MatchingEngine {
     return this.removeOrder(orderId) as StampedOwnOrder | undefined;
   }
 
-  public removePeerOrder = (orderId: string): StampedPeerOrder | undefined => {
-    return this.removeOrder(orderId) as StampedPeerOrder | undefined;
+  public removePeerOrder = (orderId: string, decreasedQuantity: number | undefined = undefined): StampedPeerOrder | undefined => {
+    const order = this.removeOrder(orderId) as StampedPeerOrder | undefined;
+
+    if (order && decreasedQuantity) {
+      order.quantity = order.quantity - decreasedQuantity;
+      this.addPeerOrder(order);
+
```

Comment:
instead of removing & adding, we can possibly just do an update.
`quantity` doesn't affect the order priority, so it shouldn't create an issue if we'll just update it. 
I was looking at the priority queue `forEach` method, but it seems to copy the entire queue before being executed, which might make it worse then removing & adding.
We might be able to do an item lookup similarly to how we currently remove items, just that we'll expose it instead of removing it. 
We can take care of this later on, but tell me if you want to open an issue for this. (@sangaman)

Reply:
I agree actually that we should be updating the existing order. We can just get the order by id and then change the quantity.